### PR TITLE
add peak velocity and peak acceleration

### PR DESCRIPTION
**Pull request by ZoeJacky**
mouse_positions:[[x0, y0], [x1, y1], ...]
mouse_times:[t0, t1, ...]

velocity:
    [[Vx0,Vy0],[Vx1,Vy1], ...]
    Vx0 = (x1-x0)/(t1-t0)
    Vy0 = (y1-y0)/(t1-t0)
    ....
acceleration:
    [[Ax0,Ay0],[Ax1,Ay1], ...]
    Ax0 = (Vx1-Vx0)/(t1-t0)
    Ay0 = (Vy1-Vy0)/(t1-t0)
    ....

peak velocity = max(velocity)
peak acceleration = max(acceleration)

### CODE REVIEW

File: src/vstt/stats.py [hunk 1]
```diff
@@ -493,3 +524,39 @@ def preprocess_mouse_positions(mouse_positions: np.ndarray) -> np.ndarray:
...
+    return velocity
+
+
+def get_acceleration(
+    mouse_times: np.ndarray, mouse_positions: np.ndarray
+) -> np.ndarray:
+    if mouse_times.size == 0 or mouse_positions.size == 0:
+        return np.array([0])
+    first_order_derivative = get_derivative(mouse_positions.transpose(), mouse_times)
+    second_order_derivative = get_derivative(first_order_derivative, mouse_times[:-1])
```
Comment (lkeegan) [inline src/vstt/stats.py:560]: You checked earlier that `mouse_times` is not empty, but if it has exactly 1 element then I think this expression will give an error - could you add a test with a single mouse_time? (also maybe it would be more robust to move the checks for empty arrays to the `get_derivative` function?)

**Review by lkeegan — COMMENTED** [review]
Could you also add definitions/descriptions of these two new statistics to docs/reference/statistics.rst?

Otherwise looks good!

File: docs/reference/statistics.rst
Comment (ZoeJacky) [inline docs/reference/statistics.rst:1]: @lkeegan Can you please help to check if the 2 math representations about velocity and acceleration are correct or not? Thank you:)

File: docs/reference/statistics.rst [hunk 2]
```diff
@@ -80,3 +80,39 @@ Given pairs of :math:`(x, y)` cursor locations, the following statistics are cal
...
+.. figure:: images/velocity.svg
+   :alt: Velocity is defined as the rate of change of position with respect to time
+
+The cursor location at a timestamp is given by a pair of :math:`(x, y)` coordinates,
+where :math:`(0, 0)` corresponds to the center of the screen, and 1 in these units is equal to the height of the screen.
+
+Given pairs of :math:`(x, y)` cursor locations, and pairs of :math:`t` timestamp, the following statistics are calculated, all in units of screen height:
+
+    * Velocity
+        * :math:`\frac{\sqrt{(x_{i+1}-x_i)^2+(y_{i+1}-y_i)^2}}{t_{i+1}-t_i}`
```
Comment (ZoeJacky) [inline docs/reference/statistics.rst:96]: @lkeegan Can you please help to check if the 2 math representations about velocity and acceleration are correct or not? Thank you:)

File: docs/reference/statistics.rst [hunk 3]
```diff
@@ -80,3 +80,39 @@ Given pairs of :math:`(x, y)` cursor locations, the following statistics are cal
...
+.. figure:: images/acceleration.png
+   :alt: Acceleration is the rate of change of the velocity of an object with respect to time
+
+The cursor location at a timestamp is given by a pair of :math:`(x, y)` coordinates,
+where :math:`(0, 0)` corresponds to the center of the screen, and 1 in these units is equal to the height of the screen.
+
+Given pairs of :math:`(x, y)` cursor locations, and pairs of :math:`t` timestamp, the following statistics are calculated, all in units of screen height:
+
+    * Acceleration
+        * :math:`\frac{\sqrt{(\frac{x_{i+2}-x_{i+1}}{t_{i+2}-t_{i+1}}-\frac{x_{i+1}-x_{i}}{t_{i+1}-t_{i}})^2+(\frac{y_{i+2}-y_{i+1}}{t_{i+2}-t_{i+1}}-\frac{y_{i+1}-y`_{i}}{t_{i+1}-t_{i}})^2}}{t_{i+1}-t_i}`
```
Comment (lkeegan) [inline docs/reference/statistics.rst:114]: extra ` in acceleration math, otherwise I think the expressions are correct

File: docs/reference/statistics.rst [hunk 4]
```diff
@@ -80,3 +80,39 @@ Given pairs of :math:`(x, y)` cursor locations, the following statistics are cal
...
+        * :math:`\frac{\sqrt{(x_{i+1}-x_i)^2+(y_{i+1}-y_i)^2}}{t_{i+1}-t_i}`
+        * the rate of change of position with respect to time
+
+    * Peak Velocity
+        * maximum velocity
+
+Acceleration
+-----------
+
+.. figure:: images/acceleration.png
```
Comment (lkeegan) [inline docs/reference/statistics.rst:105]: .png -> .svg

File: docs/reference/statistics.rst [hunk 5]
```diff
@@ -80,3 +80,39 @@ Given pairs of :math:`(x, y)` cursor locations, the following statistics are cal
...
+Velocity
+--------
+
+.. figure:: images/velocity.svg
+   :alt: Velocity is defined as the rate of change of position with respect to time
+
+The cursor location at a timestamp is given by a pair of :math:`(x, y)` coordinates,
+where :math:`(0, 0)` corresponds to the center of the screen, and 1 in these units is equal to the height of the screen.
+
+Given pairs of :math:`(x, y)` cursor locations, and pairs of :math:`t` timestamp, the following statistics are calculated, all in units of screen height:
```
Comment (lkeegan) [inline docs/reference/statistics.rst:93]: I guess the units of velocity are screen height per second

File: docs/reference/statistics.rst [hunk 6]
```diff
@@ -80,3 +80,39 @@ Given pairs of :math:`(x, y)` cursor locations, the following statistics are cal
...
+Acceleration
+-----------
+
+.. figure:: images/acceleration.png
+   :alt: Acceleration is the rate of change of the velocity of an object with respect to time
+
+The cursor location at a timestamp is given by a pair of :math:`(x, y)` coordinates,
+where :math:`(0, 0)` corresponds to the center of the screen, and 1 in these units is equal to the height of the screen.
+
+Given pairs of :math:`(x, y)` cursor locations, and pairs of :math:`t` timestamp, the following statistics are calculated, all in units of screen height:
```
Comment (lkeegan) [inline docs/reference/statistics.rst:111]: And here the units would be screen height squared per second squared

**Review by lkeegan — COMMENTED** [review]
some minor comments, otherwise looks good!

(the CI is failing due to wxPython being down, not related to your changes)

File: docs/reference/statistics.rst (hunk 6 above)
Comment (ZoeJacky) [reply docs/reference/statistics.rst:111]: Since the unit of velocity is screen height per second, the unit of acceleration should be screen height per second squared, please correct me if I'm wrong.

Comment (lkeegan) [reply docs/reference/statistics.rst:111]: yes you're right

**Review by lkeegan — APPROVED** [review]
lgtm!

**Comment by ZoeJacky** [comment]
I will merge the code to the main branch, thank you, Liam!
